File: HackerRank Solutions/Python/Algorithms/B'day Gift.py
#!/bin/python3

import os
import sys

# Complete the solve function below.
def solve(balls):
    return round(sum(balls) / 2, 1)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    balls_count = int(input())

    balls = []

    for _ in range(balls_count):
        balls_item = int(input())
        balls.append(balls_item)

    result = solve(balls)

    fptr.write(str(result) + '\n')

    fptr.close()

File: HackerRank Solutions/Java/B'day Gift.java
import java.io.*;
import java.math.*;
import java.text.*;
import java.util.*;
import java.util.regex.*;

public class Solution {

    // Complete the solve function below.
    static double solve(int[] balls) {
        double sum = 0;
        for (int b : balls) {
            sum += b;
        }
        return sum / 2;
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int ballsCount = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] balls = new int[ballsCount];

        for (int ballsItr = 0; ballsItr < ballsCount; ballsItr++) {
            int ballsItem = scanner.nextInt();
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");
            balls[ballsItr] = ballsItem;
        }

        double result = solve(balls);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedWriter.close();

        scanner.close();
    }
}
